src/pages/Overview/FiltersAndSorts.ts

```diff
@@ -1,16 +1,56 @@
-import { FILTER_ACTION_APPEND, FilterType, FilterValue } from '../../types/Filters';
-import { FAILURE, DEGRADED, HEALTHY } from '../../types/Health';
+import { ActiveFilter, FILTER_ACTION_APPEND, FilterTypeWithFilter, FilterValue } from '../../types/Filters';
+import { DEGRADED, FAILURE, HEALTHY } from '../../types/Health';
 import { NamespaceInfo } from './NamespaceInfo';
 import { SortField } from '../../types/SortFilters';
+import { MTLSStatuses } from '../../types/TLSStatus';
 
 export namespace FiltersAndSorts {
-  export const nameFilter: FilterType = {
+  export const nameFilter: FilterTypeWithFilter<NamespaceInfo> = {
     id: 'namespace_search',
     title: 'Name',
     placeholder: 'Filter by Name',
     filterType: 'text',
     action: FILTER_ACTION_APPEND,
-    filterValues: []
+    filterValues: [],
+    filter: (namespaces: NamespaceInfo[], filters: ActiveFilter[]) => {
+      return namespaces.filter(ns => filters.some(f => ns.name.includes(f.value)));
+    }
+  };
+
+  export const mtlsValues: FilterValue[] = [
+    {
+      id: 'enabled',
+      title: 'Enabled'
+    },
+    {
+      id: 'partiallyEnabled',
+      title: 'Partially Enabled'
+    },
+    {
+      id: 'notEnabled',
+      title: 'Not Enabled'
```

Comment:
Should it be rather "Disabled" or "Not Enabled" ? (I guess we have similar labels in other places, no?)

Reply:
yep, I think it is more clear to have Disabled.

Reply:
Well I see several lines below that "Not Enabled" and "Disabled" are actually two different things?

Reply:
Technically yes. It means that the NS is explicitly disabled (adding one DR and Policy). And not enabled, means that for that specific NS the user haven't changed the mTLS settings (not enabled). The difference here is made because we need to merge NS-wide mtls status and Mesh-wide mtls status. 
I think that at the end, we want to tell the user whether the ns is using mTLS or not.